Dockerfile

```diff
@@ -1,73 +1,159 @@
+# -------------------------------
+# Stage 1: Build cura-build-environment
+#
 FROM centos:centos7
 
-# Environment vars for easy configuration
-ENV CURA_BENV_BUILD_TYPE=Release
-ENV CURA_BENV_GIT_DIR=/srv/cura-build-environment
+# Set build time arugments
+ARG X11VNC_PASSWORD=123456
+ARG CURA_BUILD_ENV_BUILD_TYPE=Release
+ARG CURA_BUILD_ENV_PATH=/srv/cura-build-environment
+ARG CURA_BUILD_ENV_WORK_DIR=/tmp/cura-build-environment
+# FIXME: change to master when done
+ARG CURA_ARCUS_BRANCH_OR_TAG=WIP_fix_docker
+ARG CURA_SAVITAR_BRANCH_OR_TAG=WIP_fix_docker
+
+# Create unprivileged user
+RUN groupadd -g 1000 ultimaker \
+ && useradd -g ultimaker -u 1000 ultimaker -m
 
 # Install package repositories
-RUN yum -y update
-RUN yum install -y \
-    epel-release
-RUN yum update -y
-RUN yum install -y \
-    centos-release-scl-rh
-RUN yum update -y
-RUN yum install -y \
-    centos-release-scl
-RUN yum update -y
+RUN yum -y update && \
+    yum install -y epel-release && \
+    yum update -y && \
+    yum install -y centos-release-scl-rh centos-release-scl && \
+    yum update -y
```

Comment:
I think that we don't want to update this to be the latest version, because then our CI might suddenly break when someone releases an updated version of any library on this server. We should aim for fixed versions of every dependency we have. We can't control the source code if we don't review and test every dependency update before applying them.

Reply:
This can be split into another image as a centos base image which we build everything else upon. The base image can be built at some point of time and get assigned a unique tag (say the datetime it was built `centos7-base:20190307`, and for `cura-build-environment` and everything else we can base those on that image. This way we know that for an image like `centos7-base:20190307`, it contains all updates including security updates until that date, and we can push new updated `centos7-base` images and rebase the other ones on the new ones to include new updates in a more controlled manner.

Reply:
That sounds like a better idea.